creator-node/src/routes/files.js

```diff
@@ -170,71 +170,29 @@ const getCID = async (req, res) => {
     )
   }
 
-  startMs = Date.now()
-  const cacheKey = getStoragePathQueryCacheKey(CID)
-  let storagePath = await redisClient.get(cacheKey)
-  decisionTree.push({
-    stage: `REDIS_CLIENT_GET_CACHE_KEY`,
-    time: `${Date.now() - startMs}ms`
-  })
-
-  if (!storagePath) {
-    // Don't serve if not found in DB.
-    startMs = Date.now()
-    const queryResults = await models.File.findOne({
-      where: {
-        multihash: CID
-      },
-      order: [['clock', 'DESC']]
-    })
-    decisionTree.push({
-      stage: `DB_CID_QUERY`,
-      time: `${Date.now() - startMs}ms`
-    })
-    if (!queryResults) {
-      decisionTree.push({
-        stage: `DB_CID_QUERY_FAILED`
-      })
-      logGetCIDDecisionTree(decisionTree, req)
-      return sendResponse(
-        req,
-        res,
-        errorResponseNotFound(
-          `${logPrefix} No valid file found for provided CID`
-        )
-      )
-    }
-
-    if (queryResults.type === 'dir') {
-      decisionTree.push({
-        stage: `DB_CID_QUERY_CONFIRMED_DIR`
-      })
-      logGetCIDDecisionTree(decisionTree, req)
-      return sendResponse(
-        req,
-        res,
-        errorResponseBadRequest('this dag node is a directory')
-      )
+  const storagePath = DiskManager.computeFilePath(CID)
+  /**
+   * check if path exists
+   * if dir -> return error
+   * if file exists, continue
+   */
+  try {
+    // Will throw if path does not exist
+    // If exists, returns an instance of fs.stats class
+    const stats = await fs.stat(storagePath)
```

Comment:
I think this is going to be slow to have on the critical path. maybe this is not easy now, but i think the right way this would work would be

// path is QmXYZ or QmXYZ/480x480.jpg
getCID(path):
  try{
    const storagePath = DiskManager.computeFilePath(CID)
    const stream = await streamFromFS()
    return stream
  } catch (e) {
    try {
      const stream = findInNetwork
      return stream
    } catch (e)  {
      return e
    }
  }

Reply:
how could this be any slower than the redis + db read? should be faster no?

Reply:
i agree that it is a more heavyweight call than is needed, i'll look for alternatives

Reply:
oh lol `streamFromFileSystem` already does an fsStat, ok there's some consolidation required here